test/template-overload.html

```diff
@@ -0,0 +1,103 @@
+<!doctype html>
+<!--
+@license
+Copyright (c) 2015 The Polymer Project Authors. All rights reserved.
+This code may only be used under the BSD style license found at http://polymer.github.io/LICENSE
+The complete set of authors may be found at http://polymer.github.io/AUTHORS
+The complete set of contributors may be found at http://polymer.github.io/CONTRIBUTORS
+Code distributed by Google as part of the polymer project is also
+subject to an additional IP rights grant found at http://polymer.github.io/PATENTS
+-->
+<html>
+<head>
+  <meta charset="UTF-8">
+  <title>iron-list test</title>
+  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0">
+
+  <script src="../../webcomponentsjs/webcomponents-lite.js"></script>
+  <script src="../../web-component-tester/browser.js"></script>
+
+  <link rel="import" href="helpers.html">
+  <link rel="import" href="o-list.html">
+
+</head>
+<body>
+
+  <test-fixture id="defaultTemplateList">
+    <template>
+      <o-list></o-list>
+    </template>
+  </test-fixture>
+
+  <test-fixture id="templateOverloadList">
+    <template>
+      <o-list>
+        <template>
+          <div class="overloaded-template">[[item.index]]</div>
+        </template>
+      </o-list>
+    </template>
+  </test-fixture>
+
+  <script>
+
+    suite('template overloading', function() {
+      var container;
+
+      setup(function() {
+        container = fixture('templateOverloadList');
+      });
+
+      test('check physical item size', function(done) {
+        var setSize = 10;
+        container.items = buildDataSet(setSize);
+
+        flush(function() {
+          assert.equal(container.items.length, setSize);
+          done();
+        });
+      });
+
+      test('check item template', function(done) {
+        container.items = buildDataSet(1);
+
+        flush(function() {
+          assert.equal(container.$.list.$.items.querySelectorAll('.overloaded-template').length, 1);
+          done();
+        });
+      });
+    });
+
+    suite('default template', function() {
```

Comment:
This will make both test suites test against the <slot> being overridden, should I just add that count test to the other suite and remove this one?